pallets/xtransfer/src/bridge_transfer/mod.rs

```diff
@@ -241,38 +283,163 @@ pub mod pallet {
 		#[pallet::weight(195_000_000)]
 		pub fn transfer(
 			origin: OriginFor<T>,
-			to: T::AccountId,
+			dest: Vec<u8>,
 			amount: BalanceOf<T>,
 			rid: ResourceId,
 		) -> DispatchResult {
-			let source = T::BridgeOrigin::ensure_origin(origin)?;
-
-			if rid == T::NativeTokenResourceId::get() {
-				// ERC20 PHA transfer
-				<T as Config>::Currency::transfer(
-					&source,
-					&to,
-					amount,
-					ExistenceRequirement::AllowDeath,
-				)?;
+			let bridge_id = T::BridgeOrigin::ensure_origin(origin.clone())?;
+
+			let (src_id, dest_location) = dest.split_first().ok_or(Error::<T>::DestUnrecognised)?;
+			let u128_src_id: u128 = (*src_id)
+				.try_into()
+				.expect("Convert from chain id to u128 must be ok; qed.");
+			let src_resolve_location: MultiLocation = (
+				0,
+				X2(GeneralKey(b"solo".to_vec()), GeneralIndex(u128_src_id)),
+			)
+				.into();
+
+			let dest_location: MultiLocation = Decode::decode(&mut dest_location.clone())
+				.map_err(|_| Error::<T>::DestUnrecognised)?;
+			let dest_resolve_location: MultiLocation = dest_location
+				.clone()
+				.resolve()
+				.ok_or(Error::<T>::DestUnrecognised)?;
+
+			let asset_location: MultiLocation = Self::rid_to_location(&rid)?;
+			let asset_resolve_location: MultiLocation = asset_location
+				.clone()
+				.resolve()
+				.ok_or(Error::<T>::AssetConversionFailed)?;
+
+			log::trace!(
+				target: "BridgeTransfer.transfer",
+				"resolve location of assset ${:?}, resolve location of source: {:?}.",
+				&asset_resolve_location,
+				&src_resolve_location,
+			);
+			let imbalance = if asset_resolve_location == src_resolve_location {
```

Comment:
Have some comments added. And the `imbalance` is not actually the real imbalance, but i think it can make our implementation logic more clearly. I suggest we can just keep it here.

Reply:
I suggest to use the comment to add the context information. The code should always be clean and simple.

Reply:
Done